NuRadioReco/modules/channelSineSubtraction.py

```diff
@@ -0,0 +1,249 @@
+
+import numpy as np
+import sys
+import logging
+# Add the directory where your library is located
+#sys.path.append('/Users/nozdrina.1/Documents/phys/software/NuRadioMC')
+
+import NuRadioReco.utilities as ut
+import matplotlib.pyplot as plt
+from scipy.optimize import curve_fit
+from NuRadioReco.utilities import units, fft
+
+"""
+Contains module to filter continuous wave out of the signal using sine subtraction on peaks in frequency spectrum
+"""
+
+
+def guess_amplitude(wf: np.ndarray, target_freq: float, sampling_rate: float = 3.2):
+    """
+        Estimate the amplitude of a specific harmonic in the waveform.
+
+        Args:
+            wf (np.ndarray): Input waveform (1D array).
+            target_freq (float): Target frequency for which to estimate amplitude.
+            sampling_rate (float): Sampling rate of the waveform (GHz).
+
+        Returns:
+            float: Estimated amplitude of the target frequency.
+        """
+    if wf.size == 0:
+        raise ValueError("Input waveform is empty.")
+    if target_freq < 0 or target_freq > sampling_rate / 2:
+        raise ValueError("Target frequency is out of range (0 to Nyquist frequency).")
+
+    fft_spectrum = np.fft.fft(wf)
+    frequencies = np.fft.fftfreq(len(wf), d=1 / sampling_rate)
+
+    # Find amplitude of the 50 Hz harmonic
+
+    bin_index = np.argmin(np.abs(frequencies - target_freq))
+    amplitude = np.abs(fft_spectrum[bin_index]) / len(wf)  # Normalize
+    if bin_index > 0:
+        amplitude *= 2  # Adjust for double-sided spectrum
+    #print(f"Amplitude of {target_freq} Hz harmonic: {amplitude}")
+    return amplitude
+
+
+def sine_sub(wf: np.ndarray, sampling_rate: float = 3.2, peak_prominance: float = 6.0, saved_noise_freqs : list = None):
```

Comment:
can we change the name of the function to be a bit more self explanatory: `sinewave_subraction` for example